keps/sig-storage/1412-immutable-secrets-and-configmaps/README.md

```diff
@@ -197,6 +181,135 @@ immutable Secrets and/or ConfigMaps. That said, this is purely a performance
 improvement and doesn't have correctness implications. So those clusters will
 simple have worse scalability characteristic.
 
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: ImmutableEphemeralVolumes
+    - Components depending on the feature gate: kube-apiserver, kubelet
+
+* **Does enabling the feature change any default behavior?**
+  No, users need to explicitly opt-in for secrets/configmaps immutability.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes. kube-apiserver rollback allows for modifying secrets/configmaps.
+  kubelet rollback causes that all secrets/configmaps are again being watched.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  If the `immutable` field of Secret/ConfigMap wasn't cleared up for some
+  objects, this would result in making the immutable again.
+
+* **Are there any tests for feature enablement/disablement?**
+  Only basic conversions tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  If some critical Secrets/Configmaps were marked as immutable it may not be
+  able to update them if needed. No impact for already running workloads.
+
+* **What specific metrics should inform a rollback?**
+  No known rollback criteria.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Manual tests run to confirm that (on top of existing e2e tests):
+  - setting immutable field is impossible when the feature is switched of
+  - upgrade doesn't change behavior of existing objects
+  - immutable fields is dropped only on object touch after rollback
```

Comment:
have we verified that a Pod that sets immutable field can be deleted/updated after the apiserver is rolled back to a version with the feature disabled?

Reply:
I've repeated my tests and clearly I did something wrong previously, as this sentence isn't true.
Looking into the code it shouldn't be true.

What happens is, (which is a universal pattern used in case of all features), is that we clear the field only if the feature is disabled and also wasn't set before:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/secret/strategy.go#L87

The validation isn't gated, so it's impossible to update the data fields even after rollback. The question is whether it's desired or not - I'm not entirely sure.

I've updated the text to reflect reality and will ping Tim/Jordan for thoughts.